compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -1068,13 +1058,39 @@ impl<'a> Parser<'a> {
         Ok(TraitBoundModifiers { constness, asyncness, polarity })
     }
 
+    pub fn parse_bound_constness(&mut self) -> PResult<'a, BoundConstness> {
+        Ok(if self.eat(exp!(Tilde)) {
```

Comment:
Add a fixme here to stop accepting `~const` once the syntax is finalized and once everything switched to the new thing